docs/guide/custom-functions.md

```diff
@@ -119,72 +108,243 @@ Register your function plugin (and its translations) so that HyperFormula can re
 
 Use the [`registerFunctionPlugin()`](../api/classes/hyperformula.md#registerfunctionplugin) method:
 
-```javascript
-HyperFormula.registerFunctionPlugin(CountHF, CountHF.translations);
+```js
+HyperFormula.registerFunctionPlugin(MyCustomPlugin, MyCustomPlugin.translations);
 ```
 
 ### 6. Use your custom function inside a formula
 
-Now, you can use your HYPER function inside a formula:
+Now, you can use your GREET function inside a formula:
 
-```javascript
+```js
 // prepare spreadsheet data
-const data = [['=HYPER()']];
+const data = [['Anthony', '=GREET(A1)']];
 
 // build a HyperFormula instance where you can use your function directly
 const hfInstance = HyperFormula.buildFromArray(data);
 
 // read the value of cell A1
-const result = hfInstance.getCellValue({ sheet: 0, col: 0, row: 0 });
+const result = hfInstance.getCellValue({ sheet: 0, col: 1, row: 0 });
 
 // open the browser's console to see the results
 console.log(result);
 ```
 
 ### Full example
 
-```javascript
-import { FunctionPlugin } from 'hyperformula';
+The complete implementation of this custom function is also included in the [demo](#working-demo).
 
-export class CountHF extends FunctionPlugin {
-  hyper(ast, state) {
-    return this.runFunction(ast.args, state, this.metadata('HYPER'), () => 'Hyperformula'.length);
+```js
+import { FunctionPlugin, CellError } from 'hyperformula';
```

Comment:
`CellError` was not used up to this point, so it looks excessive. However, I think that a simple cell error should be added to the snippet in point "4. Implement your function's logic"

Reply:
Initially, I had a simple error-handling example in the implementation of `GREET`, but I removed it to keep the first part of the tutorial simple. Error handling is presented in the `Advanced custom function example` section. Do you think, the first part should also contain an example of `return new CellError(...);`?

Reply:
I think yes, since it is only three lines more. It teaches something interesting about the type handling

Reply:
Done